src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -55,76 +57,93 @@ const StudentApplication = ({ userProfile, authentication }) => {
-      const initialApplicants = [{ Profile: userProfile, OffCampusProgram: '' }];
-      setUnsavedChanges(true);
+      const initialApplicants = [
+        { Profile: userProfile, Username: userProfile.AD_Username, OffCampusProgram: '' },
+      ];
       setApplicationDetails({
         ...BLANK_APPLICATION_DETAILS,
         EditorProfile: userProfile,
         Gender: userProfile.Gender,
         Applicants: initialApplicants,
       });
       setCanEditApplication(true);
+      setUnsavedChanges(true);
     };
 
-    /**
-     * Load the user's saved apartment application, if one exists
-     *
-     * @async
-     * @function loadApplication
-     */
-    const loadApplication = async () => {
-      try {
-        setLoading(true);
-        // Check if the current user is on an application. Returns the application ID number if found
-        const newApplicationID = await housing.getCurrentApplicationID();
-        if (newApplicationID === null || newApplicationID === -1) {
-          initializeNewApplication();
-        } else {
-          setUnsavedChanges(false);
-          const newApplicationDetails = await housing.getApartmentApplication(newApplicationID);
-          if (newApplicationDetails) {
-            setApplicationDetails(newApplicationDetails);
-            setCanEditApplication(
-              userProfile.AD_Username === newApplicationDetails.EditorProfile.AD_Username ?? false,
-            );
-            setUnsavedChanges(false);
-          }
-        }
-      } catch (error) {
+    let result = false;
+    try {
+      // Check if the current user is on an application. Returns the application ID number if found
+      const newApplicationID = await housing.getCurrentApplicationID();
+      if (newApplicationID > 0) {
+        const newApplicationDetails = await housing.getApartmentApplication(newApplicationID);
+        setApplicationDetails(newApplicationDetails);
+        debugPrintApplicationDetails(newApplicationDetails);
+        setCanEditApplication(
+          userProfile.AD_Username === newApplicationDetails.EditorProfile.AD_Username ?? false,
+        );
+        setUnsavedChanges(false);
+        result = true;
+      } else {
+        throw createError(new Error('Invalid application ID'), { status: 404 });
+      }
+    } catch (e) {
+      if (e instanceof NotFoundError) {
         initializeNewApplication();
-      } finally {
-        setLoading(false);
+      } else if (e instanceof AuthError) {
+        const debugMessage =
+          'Received a 401 (Unauthorized) response, which should not be possible in this case. Please try refreshing the page, or contact CTS for support.';
+        console.error(`Debug Message: ${debugMessage}`);
+        createSnackbar(debugMessage, 'error');
+      } else {
+        throw e;
       }
-    };
-
-    loadApplication();
+    } finally {
+      setNewEditorProfile(null);
+      return result;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
When testing locally, `eslint` wasn't warning me about missing dependencies. Was this left over from testing? It is almost never a good idea to disable that warning because it warns about some potential bugs that are VERY hard to track down otherwise.

Reply:
The eslint was simply catching the call to `debugPrintApplicationDetails`, which was intentionally excluded

If I have since removed the call to `debugPrintApplicationDetails`, then I must have forgotten to remove this `// eslint-disable-next-line`